packages/snap/src/index.ts

```diff
@@ -1,33 +1,92 @@
-import { OnRpcRequestHandler } from '@metamask/snaps-types';
-import { panel, text } from '@metamask/snaps-ui';
+import { providerErrors, rpcErrors } from '@metamask/rpc-errors';
+import { DialogType, OnRpcRequestHandler } from '@metamask/snaps-types';
+import { copyable, heading, panel, text } from '@metamask/snaps-ui';
+import { SLIP10Node } from '@metamask/key-tree';
+import {
+  add0x,
+  assert,
+  bytesToHex,
+  remove0x,
+  valueToBytes,
+} from '@metamask/utils';
+import { sign as signEd25519 } from '@noble/ed25519';
+import { sign as signSecp256k1 } from '@noble/secp256k1';
+
+import type { GetBip32PublicKeyParams, SignMessageParams } from './types';
 
 /**
  * Handle incoming JSON-RPC requests, sent through `wallet_invokeSnap`.
  *
  * @param args - The request handler args as object.
- * @param args.origin - The origin of the request, e.g., the website that
- * invoked the snap.
  * @param args.request - A validated JSON-RPC request object.
- * @returns The result of `snap_dialog`.
+ * @returns The result of the method.
  * @throws If the request method is not valid for this snap.
  */
-export const onRpcRequest: OnRpcRequestHandler = ({ origin, request }) => {
+export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
   switch (request.method) {
-    case 'hello':
-      return snap.request({
+    case 'getPublicKey':
+      return await snap.request({
+        method: 'snap_getBip32PublicKey',
+        params: request.params as unknown as GetBip32PublicKeyParams,
+      });
+
+    case 'signMessage': {
+      const { message, curve, ...params } = request.params as SignMessageParams;
+      const json = await snap.request({
+        method: 'snap_getBip32Entropy',
+        params: {
+          ...params,
+          curve,
+        },
+      });
+      const node = await SLIP10Node.fromJSON(json);
+
+      assert(node.privateKey);
+      assert(curve === 'ed25519' || curve === 'secp256k1');
```

Comment:
This logic looks plausible to me, but I'm not familiar with the APIs we're using. Do you have a doc link? If so, let's also add it as a comment

Reply:
Sounds good, I'll add a comment. The SLIP-10 supports NIST P-256, but Metamask currently doesn't (under the assumption that there isn't enough demand to justify the support). Besides NIST P-256, SLIP-10 supports only ed25519 and secp256k1.

The link will be this one
https://github.com/satoshilabs/slips/blob/master/slip-0010.md#master-key-generation